### PR TITLE
RUI1001

### PR DESCRIPTION
**Pull request by fcbjimmy**
RUI1001 - [3] - Merge Authentication and Todo List

https://trello.com/c/05HchQ7i

- Merge Authentication and Todo app
- Refactored todo app code

Blockers: name showing 'null' when its logged in at first after signing up

### CODE REVIEW

File: src/components/TodoApp/TodoApp.js [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+      return JSON.parse(savedTodos);
+    }
+    return [];
+  });
+
+  useEffect(() => {
+    localStorage.setItem('todos', JSON.stringify(todos));
+  }, [todos]);
+
+  function addTodo(newtodo) {
```
Comment (cheungjaidev) [inline src/components/TodoApp/TodoApp.js:26]: Refactor needed.

File: src/containers/Index.jsx [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+
+function SignInOutContainer() {
+  const [value, setValue] = useState(1);
+
+  const handleChange = (event, newValue) => {
+    setValue(newValue);
+    console.log(newValue);
+  };
+
+  console.log({ value });
```
Comment (cheungjaidev) [inline src/containers/Index.jsx]: remove

**Review by cheungjaidev — CHANGES_REQUESTED** [review]

File: src/containers/Index.jsx (hunk 2 above)
Comment (fcbjimmy) [reply src/containers/Index.jsx]: removed

File: src/components/authentication/Welcome.jsx [hunk 3]
```diff
@@ -0,0 +1,27 @@
+import React, { useEffect, useState } from 'react';
+import { Typography } from '@mui/material';
+import { useUserContext } from '../../context/userContext';
+
```
Comment (yutooo5225) [inline src/components/authentication/Welcome.jsx:4]: conside to use CSS:
```
text-transform: capitalize;
```

File: src/components/authentication/Login.jsx [hunk 4]
```diff
@@ -0,0 +1,148 @@
...
+    password: '',
+    showPassword: false,
+  });
+
+  const handleChangePass = (prop) => (event) => {
+    setValues({ ...values, [prop]: event.target.value });
+  };
+
+  const handleClickShowPassword = () => {
+    setValues({ ...values, showPassword: !values.showPassword });
```
Comment (yutooo5225) [inline src/components/authentication/Login.jsx:44]: if you want to change the state depend on *previous state*, use arrow function instead, i.e. `prev => !prev`

**Pull request merged**